components/canvas/canvas_paint_task.rs

```diff
@@ -301,8 +250,25 @@ impl<'a> CanvasPaintTask<'a> {
     }
 
     fn fill_rect(&self, rect: &Rect<f32>) {
-        self.drawtarget.fill_rect(rect, self.state.fill_style.to_pattern_ref(),
-                                  Some(&self.state.draw_options));
+        if self.need_to_draw_shadow() {
+            let source_rect = self.state.transform.transform_rect(rect);
+            let new_draw_target = self.create_draw_target_for_shadow(&source_rect);
+
+            new_draw_target.fill_rect(rect, self.state.fill_style.to_pattern_ref(),
+                                      Some(&self.state.draw_options));
+
+            self.drawtarget.draw_surface_with_shadow(new_draw_target.snapshot(),
+                                                     &Point2D(source_rect.origin.x as AzFloat,
+                                                              source_rect.origin.y as AzFloat),
+                                                     &self.state.shadow_color,
+                                                     &Point2D(self.state.shadow_offset_x as AzFloat,
+                                                              self.state.shadow_offset_y as AzFloat),
+                                                     (self.state.shadow_blur / 2.0f64) as AzFloat,
```

Comment:
The shadowBlur attribute specifies the level of the blurring effect.
And the sigma as the standard deviation to perform a 2D Gaussian blur is half the value of shadowBlur.
Please refer to the 4-1 on the spec https://html.spec.whatwg.org/multipage/#when-shadows-are-drawn.